src/main/java/com/marcusslover/plus/lib/sound/Music.java

```diff
@@ -0,0 +1,213 @@
+
+/**
+ * Music can be used to loop notes for an audience or command sender.
+ *
+ * <p>Example:
+ * <pre>
+ *     {@code
+ *     Music.of(Note.of(Key.key("music.loop"), 0.2f, 1f), 2400).send(audience);
+ *     }
+ * </pre>
+ * This will play the note "music.loop" for 2400 ticks (2 minutes) for the audience.
+ * <p>
+ */
+@Data
+@Accessors(fluent = true)
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public class Music implements ISendable<Music> {
+	protected final Note intro;
+	protected final long introLength;
+	protected final Note loop;
+	protected final long loopLength;
+	protected final Note tail;
+
+	protected Task loopTask = null;
+	public long ticks = 0;
+	protected Task tickTask = Task.asyncRepeating(ServerUtils.getCallingPlugin(), () -> {
+		ticks++;
+	}, 0, 1);
+	public int loops = 0;
+
+
+	/**
+	 * Stop the music for an audience and play the tail if it exists.
+	 *
+	 * @param audience The audience.
+	 */
+	public void stop(Audience audience) {
+		this.tickTask.cancel();
+		this.loopTask.cancel();
+		// If we are playing the loop sound, then wait until its finished before then playing the tail.
+		if (this.loops > 0) {
+			Task.syncDelayed(ServerUtils.getCallingPlugin(), () -> {
+				audience.stopSound(this.loop.sound);
+				if (this.tail != null) {
+					this.tail.send(audience);
+				}
+			}, this.loopLength - (this.ticks % this.loopLength));
+		} else {
+			audience.stopSound(this.loop.sound);
+			if (this.tail != null) {
+				this.tail.send(audience);
+			}
+		}
+	}
+
+	/**
+	 * Stop the music for a command sender and play the tail if it exists.
+	 *
+	 * @param target The command sender.
+	 */
+	public void stop(CommandSender target) {
+		this.tickTask.cancel();
+		this.loopTask.cancel();
+		if (this.loops > 0) {
+			Task.syncDelayed(ServerUtils.getCallingPlugin(), () -> {
+				target.stopSound(this.loop.sound);
+				if (this.tail != null) {
+					this.tail.send(target);
+				}
+			}, this.loopLength - (this.ticks % this.loopLength) - 1);
+		} else {
+			target.stopSound(this.loop.sound);
+			if (this.tail != null) {
+				this.tail.send(target);
+			}
+		}
+	}
+
+	/**
+	 * Stop the music from the audience without playing the tail.
+	 *
+	 * @param audience The audience.
+	 */
+	public void stopAll(Audience audience) {
+		this.tickTask.cancel();
+		this.loopTask.cancel();
+		if (this.intro != null)
+			audience.stopSound(this.intro().sound);
+		if (this.loop != null)
+			audience.stopSound(this.loop().sound);
+		if (this.tail != null)
+			audience.stopSound(this.tail().sound);
+	}
+
+	/**
+	 * Stop the music from the command sender without playing the tail.
+	 *
+	 * @param target The command sender.
+	 */
+	public void stopAll(CommandSender target) {
+		this.tickTask.cancel();
+		this.loopTask.cancel();
+		if (this.intro != null)
+			target.stopSound(this.intro().sound);
+		if (this.loop != null)
+			target.stopSound(this.loop().sound);
+		if (this.tail != null)
+			target.stopSound(this.tail().sound);
```

Comment:
Fails to follow code style of this repo.

Style: https://google.github.io/styleguide/javaguide.html